### PR TITLE
BUG: Let PdfWriter.pages return PageObject after calling `clone_document_from_reader()`

### LINKED ISSUE #1595: page 2 blended with page 1 on output via append
I need to read in a fillable form pdf, fill it out, then generate result pdf. this is largely working, except that page 2 of the pdf (which contains only text) doesn't get properly appended. It's text contents are laid over the top of a copy of page 1. 

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
macOS-11.6.5-x86_64-i386-64bit

$ python -c "import pypdf;print(pypdf.__version__)"
3.3.0
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
from pypdf import PdfReader, PdfWriter

reader = PdfReader('cms1500.pdf')
writer = PdfWriter()
writer.append(reader)

with open('copy.pdf', "wb") as output_stream:
    writer.write(output_stream)
```

Pdf source is 
[form-cms1500.pdf](https://github.com/py-pdf/pypdf/files/10564431/form-cms1500.pdf) - a standard pdf in healthcare. There should be no problem with you using it in test suites.

## Results

Outcome from using `writer.append`:
Screen Shot 2023-02-01 at 10 21 22 PM

Outcome from using `writer.append(reader, [1])`
Screen Shot 2023-02-01 at 10 22 48 PM

NOTE: Outcome from using `writer.clone_document_from_reader` does produce expected output **BUT i can't use it as it causes other problems (namely after clone writer.pages items appear to be DictionaryObject and not PageObject)**:
Screen Shot 2023-02-01 at 10 24 26 PM

### PR DESCRIPTION
**Pull request by pubpub-zz**
Currently, the PdfWriter.pages attribute returns a `DictionartyObject`. With this fix, the attribute will return Iterable[PageObject].

Fixes #1595

See also:
* #1650
* #1654

### CODE REVIEW

File: pypdf/_writer.py [hunk 1]
```diff
@@ -852,6 +852,11 @@ def clone_reader_document_root(self, reader: PdfReader) -> None:
         self._root = self._root_object.indirect_reference  # type: ignore[assignment]
         self._pages = self._root_object.raw_get("/Pages")
         self._flatten()
+        for p in self.flattened_pages:
+            o = self._objects[p.idnum - 1].get_object()
+            self._objects[p.idnum - 1] = PageObject(self, p)
```
Comment (MartinThoma) [inline pypdf/_writer.py:857]: Why did you add this line?

File: pypdf/_writer.py [hunk 2]
```diff
@@ -852,6 +852,11 @@ def clone_reader_document_root(self, reader: PdfReader) -> None:
...
+            for k, v in o.items():
+                self._objects[p.idnum - 1][k] = v
```
Comment (MartinThoma) [inline pypdf/_writer.py:859]: Does this loop add all elements of the page to the writer as indirect objects? Why is this necessary?

File: pypdf/_writer.py (hunk 1 above)
Comment (pubpub-zz) [reply pypdf/_writer.py:857]: do you mean L857 ? if it is to create Pages. The cloning process works at low level and duplicates an DictionaryObject into a DictionaryObject.

File: pypdf/_writer.py (hunk 2 above)
Comment (pubpub-zz) [reply pypdf/_writer.py:859]: reviewing the PR I'v slighty simplified the code.

**Comment by MartinThoma** [comment]
@pubpub-zz Could you maybe explain the title `BUG: pages[] return PageObject after calling clone_document_from_reader()` to me?

Does it mean that before this PR, the `PdfWriter.pages was returning a single PageObject after calling `clone_document_from_reader`  and now it returns a list?

**Comment by pubpub-zz** [comment]
currently in `PdfWriter pages[]` returns `DictionartyObject` and not `PageObjects`

**Comment by MartinThoma** [comment]
Thank you!
